### ISSUE TITLE
unsatisfiable classes when merged into vertebrate.owl

### ISSUE THREAD

**Issue by RLWOHIO** [obophenotype/zebrafish-phenotype-ontology#36]
The latest release (2019-11-12) is a big improvement over the previous version.  When standing alone, all classes are satisfiable.  After being incorporated into vertebrate.owl though, some of them became unsatisfied, as listed below:
ZFA_0000435
ZP_0000412
ZP_0002062
ZP_0002510
ZP_0002511
ZP_0002557
ZP_0003019
ZP_0003144
ZP_0003145
ZP_0003428
ZP_0004008
ZP_0004283
ZP_0004284
ZP_0004299
ZP_0005361
ZP_0005362
ZP_0005531
ZP_0005928
ZP_0006699
ZP_0006700
ZP_0006899
ZP_0006921
ZP_0006922
ZP_0008170
ZP_0008184
ZP_0008185
ZP_0008312
ZP_0008768
ZP_0009148
ZP_0010431
ZP_0010686
ZP_0010687
ZP_0010688
ZP_0011448
ZP_0011473
ZP_0011681
ZP_0011709
ZP_0011716
ZP_0011745
ZP_0011889
ZP_0012376
ZP_0012463
ZP_0012500
ZP_0013387
ZP_0014696
ZP_0014741
ZP_0014762
ZP_0014898
ZP_0015602
ZP_0016403
ZP_0016404
ZP_0016405
ZP_0016733
ZP_0016944
ZP_0016945
ZP_0016946
ZP_0017766
ZP_0018215
ZP_0018216
ZP_0018220
ZP_0018289
ZP_0018331
ZP_0019846
ZP_0019851
ZP_0019852
ZP_0019853
ZP_0019854
ZP_0020515
ZP_0020776
ZP_0020778
ZP_0021475
ZP_0021924
ZP_0022946
ZP_0100778
ZP_0100779
ZP_0100967
ZP_0102325
ZP_0102981
ZP_0102982
ZP_0103341
ZP_0103347
ZP_0103558
ZP_0103989
ZP_0106021
ZP_0107100
ZP_0107101
ZP_0108328

**Comment by matentzn** [comment]
Thanks! Some of the unsatisfiablities may only be solved much later in the year! Feel free to drop me an email so I can tell you about the general trajectory of uPheno and ZP -> we are working hard on pushing for a fresh, conflict free Unified Phenotype ontology by the end of the month, and I would love to get your input/requirements if you have any! :) vertebrate.owl will hopefully be deprecated before the summer.

**Comment by RLWOHIO** [comment]
This is great to know.  Very much looking forward to having an error-free
vertebrate.owl replacement.  Since my work is primarily fish oriented, ZP
and improved mapping of ZP terms to zebrafish genes are of great
importance.  I will continue to provide feedback as issues come up down the
road.  Thanks so much for your effort!

**Comment by matentzn** [comment]
Can you describe a bit what you are using vertebrate.owl for that you cant get from simply using ZP? Which table to you use for the gene:phenotype annotations?

**Comment by RLWOHIO** [comment]
Vertebrate.owl serves as the foundation of semantic analysis of chemical
toxicities. While my target species are fish, I have broad interest on
other vertebrates, particularly human and mouse, and a range of biological
domains as well.  My last use (a year ago) of zf gene:phenotype
associations came from
https://github.com/obophenotype/zebrafish-phenotype-ontology-build.  It
seems to have been deprecated since.

**Comment by matentzn** [comment]
So basically you are looking for a phenotype ontology that integrates all vertebrate phenotypes (and their anatomy, gene functions etc)? How exactly do you use vertrebrate.owl if you dont mind me asking? In a software tool? Or just to retrieve stuff using Protege?

**Comment by RLWOHIO** [comment]
Yes.  I use a java application I put together based on the OWAPI,
reasoners, and the Semantic Measure Library (
https://www.semantic-measures-library.org/sml/).  Thanks!

**Comment by matentzn** [comment]
Can you describe your project in a bit more detail? Is your code on GitHub? What do you need from vertebrate.owl that you cant get from zp.owl all by itself?

**Comment by RLWOHIO** [comment]
My apologies for this delayed response.  Our general interest is in the
realm of mapping chemical-induced toxicity responses, particularly those of
non-molecular nature, to their biology: genes, pathways, and diseases.
This is why ontology-based semantic analysis is such an useful tool. While
our focus is on fish and zp is great, having vertebrates included gives us
a much wider coverage of biology and greater utilization of phenomics data
available.  The Java application is currently not available on Github. It
is based on OWLAPI, Semantic Measure Library (
https://www.semantic-measures-library.org/sml/), and various reasoners, all
publicly available.

**Comment by matentzn** [comment]
We are developing a new version of the Unified Phenotype Ontology - this will eventually replace vertebrate.owl and the rest of upheno1; I would be interested in your feedback and see whether it covers your use cases. Its not published yet, but a test version is available [here](https://ci.monarchinitiative.org/view/pipelines/job/upheno2/81/artifact/src/curation/upheno-release/all/upheno_all_with_relations.owl). Please let me know what you think and what is missing from in there (or too much) to cover your case.

**Comment by RLWOHIO** [comment]
This is great.  I would love to try it out when I get a chance.   Thanks a
lot for the offer.

**Comment by RLWOHIO** [comment]
I have done some testing.  On its own, upheno_all_with_relations.owl (03/04/2020), is fine.  Once merged with my annotations and relevant imports by OntoFox, there was a total 35 unsatisfiable classes, all from ZP.  This is a huge improvement over vertebrate.owl dated Dec 2019 which had a total of 455 unsatisfiable classes including 340 ZPs. Please see the attached for details.  Hope this helps.
[upheno_all_with_relations.owl.PLUS.mine2.owl.unsatisfiableZPs.pptx](https://github.com/obophenotype/zebrafish-phenotype-ontology/files/4325046/upheno_all_with_relations.owl.PLUS.mine2.owl.unsatisfiableZPs.pptx)

**Comment by matentzn** [comment]
Ah very cool. So it seems that all the phenotypes that are unsatisfiable are so because of ZFA, which in turn interacts badly with UBERON. Can you delete the axiom:

![image](https://user-images.githubusercontent.com/7070631/76547486-260c7500-6485-11ea-84e6-e6c731991670.png)

And check whether that is the problem? If so, I will talk to RO and UBERON and we will decide on how to fix it.

**Comment by RLWOHIO** [comment]
Yes, you are right.  By eliminating ZFA_0001284 and ZFA_0005491 from the
merged ontology, all the other previously unsatisfiable ZP classes became
satisfiable.
